metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/GrokController.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.rest.controller;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import org.apache.metron.rest.RestException;
+import org.apache.metron.rest.model.GrokValidation;
+import org.apache.metron.rest.service.GrokService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/v1/grok")
+public class GrokController {
+
+    @Autowired
+    private GrokService grokService;
+
+    @ApiOperation(value = "Applies a Grok statement to a sample message")
+    @ApiResponse(message = "JSON results", code = 200)
+    @RequestMapping(value = "/validate", method = RequestMethod.POST)
+    ResponseEntity<GrokValidation> post(@ApiParam(name = "grokValidation", value = "Object containing Grok statement and sample message", required = true) @RequestBody GrokValidation grokValidation) throws RestException {
```

Comment:
This doesn't quite match how a grok statement should be structured but it did uncover a bug.  The original implementation assumed the grok statement would only consist of 1 line and extracted the pattern label as the first word in that line.  If the grok statement consists of multiple lines this service would use the first word of the first line as the pattern label which is obviously not correct.

So I changed this to match how the GrokParser works:  a statement can consist of multiple lines and the pattern label is passed in as a parameter.  In your test case the request would look like this (statement is now 2 lines):

{
  "sampleData": "301dd766-78c6-4c56-b8c7-178bb91fdb2c",
  "statement": "UUID [A-Fa-f0-9]{8}-(?:[A-Fa-f0-9]{4}-){3}[A-Fa-f0-9]{12}\nLABEL %{UUID:id}",
  "patternLabel":"LABEL"
}

and the result would be:

{
  "patternLabel": "LABEL",
  "statement": "UUID [A-Fa-f0-9]{8}-(?:[A-Fa-f0-9]{4}-){3}[A-Fa-f0-9]{12}\nLABEL %{UUID:id}",
  "sampleData": "301dd766-78c6-4c56-b8c7-178bb91fdb2c",
  "results": {
    "id": "301dd766-78c6-4c56-b8c7-178bb91fdb2c"
  }
}